include/dali/core/tensor_shape.h

```diff
@@ -637,6 +638,44 @@ struct TensorListShapeBase {
     shapes.resize(num_samples * sample_dim);
   }
 
+
+  ///@{
+  /**
+   * Append this TensorListShape with shapes from other TensorListShapes.
+   */
+  void append(span<const TensorListShape<sample_ndim>> tlss) {
+    int new_samples = std::accumulate(tlss.begin(), tlss.end(), 0,
+                                      [](int acc, const auto &tls) -> int {
+                                          return acc + tls.num_samples();
+                                      });
+    if (new_samples == 0) return;
+    auto reference_dim_value = empty() ? tlss[0].sample_dim() : sample_dim();
+    for (const auto &tls : tlss) {
+      if (tls.sample_dim() != reference_dim_value) {
+        throw std::invalid_argument(
+                "All samples in TensorListShape must have the same number of dimensions");
+      }
+    }
+    nsamples += new_samples;
+    if (sample_dim() <= 0) {
+      // If `this` TLS has DynamicDimensions and wasn't initialized yet
+      set_sample_dim(tlss[0].sample_dim());
+    }
+    shapes.reserve(nsamples * sample_dim());
+    for (const auto &tls : tlss) {
+      shapes.insert(shapes.end(), tls.shapes.begin(), tls.shapes.end());
+    }
+  }
+
+  void append(const TensorListShape<sample_ndim> &tls) {
+    append(make_cspan({tls}));
```

Comment:
```suggestion
    append(make_cspan(&tls, 1));
```
Otherwise you'd be copying `tls` to an `initializer_list`.

Reply:
Done